Mathlib/Order/RelClasses.lean

```diff
@@ -0,0 +1,811 @@
+  IsWellFounded.fix (· > ·)
+
+/-- The value from `WellFoundedGt.fix` is built from the successive ones as specified. -/
+theorem fix_eq {C : α → Sort _} (F : ∀ x : α, (∀ y : α, x < y → C y) → C x) :
+    ∀ x, fix F x = F x fun y _ => fix F y :=
+  IsWellFounded.fix_eq _ F
+
+/-- Derive a `WellFoundedRelation` instance from a `WellFoundedGt` instance. -/
+def toWellFoundedRelation : WellFoundedRelation α :=
+  IsWellFounded.toWellFoundedRelation (· > ·)
+
+end WellFoundedGt
+
+/-- Construct a decidable linear order from a well-founded linear order. -/
+noncomputable def IsWellOrder.LinearOrder (r : α → α → Prop) [IsWellOrder α r] : LinearOrder α :=
+  letI := fun x y => Classical.dec ¬r x y
+  linearOrderOfSTO r
+
+/-- Derive a `WellFoundedRelation` instance from a `IsWellOrder` instance. -/
+def IsWellOrder.toHasWellFounded [LT α] [hwo : IsWellOrder α (· < ·)] : WellFoundedRelation α where
+  rel := (· < ·)
+  wf := hwo.wf
+
+-- This isn't made into an instance as it loops with `is_irrefl α r`.
+theorem Subsingleton.isWellOrder [Subsingleton α] (r : α → α → Prop) [hr : IsIrrefl α r] :
+    IsWellOrder α r :=
+  { hr with
+    trichotomous := fun a b => Or.inr <| Or.inl <| Subsingleton.elim a b,
+    trans := fun a b _ h => (not_rel_of_subsingleton r a b h).elim,
+    wf := ⟨fun a => ⟨_, fun y h => (not_rel_of_subsingleton r y a h).elim⟩⟩ }
+
+instance [Subsingleton α] : IsWellOrder α EmptyRelation :=
+  Subsingleton.isWellOrder _
+
+instance (priority := 100) [IsEmpty α] (r : α → α → Prop) : IsWellOrder α r where
+  trichotomous := isEmptyElim
+  trans := isEmptyElim
+  wf := well_founded_of_empty r
+
+instance [IsWellFounded α r] [IsWellFounded β s] : IsWellFounded (α × β) (Prod.Lex r s) :=
+  ⟨(Prod.lex (IsWellFounded.toWellFoundedRelation _) (IsWellFounded.toWellFoundedRelation _)).wf⟩
+
+instance [IsWellOrder α r] [IsWellOrder β s] : IsWellOrder (α × β) (Prod.Lex r s) where
+  trichotomous := fun ⟨a₁, a₂⟩ ⟨b₁, b₂⟩ =>
+    match @trichotomous _ r _ a₁ b₁ with
+    | Or.inl h₁ => Or.inl <| Prod.Lex.left _ _ h₁
+    | Or.inr (Or.inr h₁) => Or.inr <| Or.inr <| Prod.Lex.left _ _ h₁
+    | Or.inr (Or.inl (.refl _)) =>
+        match @trichotomous _ s _ a₂ b₂ with
+        | Or.inl h => Or.inl <| Prod.Lex.right _ h
+        | Or.inr (Or.inr h) => Or.inr <| Or.inr <| Prod.Lex.right _ h
+        | Or.inr (Or.inl (.refl _)) => Or.inr <| Or.inl rfl
+  trans a b c h₁ h₂ := by
+    cases' h₁ with a₁ a₂ b₁ b₂ ab a₁ b₁ b₂ ab <;> cases' h₂ with _ _ c₁ c₂ bc _ _ c₂ bc
+    · exact Prod.Lex.left _ _ (trans ab bc)
+    · exact Prod.Lex.left _ _ ab
+    · exact Prod.Lex.left _ _ bc
+    · exact Prod.Lex.right _ (trans ab bc)
+  wf := (Prod.lex (IsWellFounded.toWellFoundedRelation _)
+    (IsWellFounded.toWellFoundedRelation _)).wf
+
+instance (r : α → α → Prop) [IsWellFounded α r] (f : β → α) : IsWellFounded _ (InvImage r f) :=
+  ⟨InvImage.wf f IsWellFounded.wf⟩
+
+-- TODO: How to align this?
+-- it was `⟨measure_wf f⟩` before
```

Comment:
Just don't align?

Reply:
Basically, the question is, in places where `measure_wf f` is used, now one would use `(measure \.).wf`. Ideally, that transformation is something that mathport can handle.